packages/tackle-box/lib/components/Input/Input.tsx

```diff
@@ -0,0 +1,91 @@
+import { useState } from "react";
+import { html, css } from "react-strict-dom";
+import { ComponentProps } from "react";
+import { sizing, colors } from "../../vars/index.stylex";
+
+const styles = css.create({
+  container: {
+    position: "relative",
+  },
+  input: {
+    backgroundColor: colors.background,
+    borderWidth: 1,
+    borderRadius: sizing[2],
+    borderStyle: "solid",
+    borderColor: colors.border,
+    color: colors.textPrimary,
+    fontSize: sizing[4],
+    paddingInline: sizing[4],
+    paddingTop: sizing[4],
+    paddingBottom: sizing[4],
+  },
+  inputWithValue: {
+    paddingBottom: sizing[2],
+    paddingTop: sizing[6],
+  },
+  inputDisabled: {
+    backgroundColor: colors.backgroundDisabled,
+    color: colors.textDisabled,
+    cursor: "not-allowed",
+  },
+  label: {
+    color: colors.textSecondary,
+    fontSize: sizing[4],
+    left: sizing[4],
+    pointerEvents: "none",
+    position: "absolute",
+    top: sizing[4],
+    transform: "translateY(0) scale(1)",
+    transformOrigin: "left top",
+    transitionProperty: "transform",
+    transitionDuration: "200ms",
+    transitionTimingFunction: "cubic-bezier(0.4, 0, 0.2, 1)",
+  },
+  labelRaised: {
+    transform: `translateY(-6px) scale(0.75)`,
+  },
+  labelDisabled: {
+    color: colors.textDisabled,
+  },
+});
+
+type InputProps = ComponentProps<typeof html.input>;
+
+type Props = {
+  label: string;
+  value: string;
+  onChange: (value: string) => void;
+} & Omit<InputProps, "onChange" | "value">;
```

Comment:
idea: We could make a `MergeProps<InputProps, { foo: number }>` type where the keys in the 2nd argument are automatically omitted from the first if they exist on the first.

Reply:
I like it - added